code/modules/mob/living/carbon/carbon.dm

```diff
@@ -500,17 +504,19 @@ GLOBAL_LIST_INIT(ventcrawl_machinery, list(/obj/machinery/atmospherics/unary/ven
 						to_chat(src, "<span class='warning'>You can't crawl around in the ventilation ducts with items!</span>")
 						return
 
-			visible_message("<b>[src] scrambles into the ventilation ducts!</b>", "You climb into the ventilation system.")
-			src.loc = vent_found
-			add_ventcrawl(vent_found)
-
-	else
-		to_chat(src, "<span class='warning'>This ventilation duct is not connected to anything!</span>")
+			if(istype(src, /mob/living/silicon/robot/drone) && !entrance_found.can_crawl_through())
+				visible_message("<span class='boldnotice'>[src] scrambles into the ventilation ducts! With a resounding snap the [entrance_found] is fastened back in place.</span>", \
+				"<span class='notice'>You climb into the ventilation system. With a resounding snap the [entrance_found] is fastened back in place.</span>")
+			else
+				visible_message("<span class='boldnotice'>[src] scrambles into the ventilation ducts!</span>", "<span class='notice'>You climb into the ventilation system.</span>")
+			loc = entrance_found
```

Comment:
```suggestion
			forceMove(entrance_found)
```
Will make sure other logic won't break

Reply:
I tried this. It completely breaks ventcrawling, by making all nonpipes solid black sprites.